ir/pattern/conjunction.rs

```diff
@@ -55,6 +55,14 @@ impl Conjunction {
         &mut self.nested_patterns
     }
 
+    pub fn optimise_away(&mut self) {
```

Comment:
`set_unsatisfiable` the fact it replaces itself with a different constraint, or sets a boolean or whatever, is implementation detail.